src/dashboard/src/components/archival_storage/tests/test_archival_storage.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+
+# This file is part of Archivematica.
+#
+# Copyright 2010-2019 Artefactual Systems Inc. <http://artefactual.com>
+#
+# Archivematica is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Archivematica is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Archivematica.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+import json
+import os
+
+from django.http import HttpResponse, HttpResponseNotFound, StreamingHttpResponse
+import pytest
+
+from components.archival_storage import atom
+from components import helpers
+import elasticSearchFunctions
+import storageService as storage_service
+
+import metsrw
+
+
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+CONTENT_DISPOSITION = "Content-Disposition"
+CONTENT_TYPE = "Content-Type"
+
+
+@pytest.fixture
+def mets_document():
+    return metsrw.METSDocument.fromfile(
+        os.path.join(THIS_DIR, "fixtures", "mets_two_files.xml")
+    )
+
+
+def test_load_premis(mets_document):
+    """Test the load of PREMIS 2 or PREMIS 3 from a given METS document."""
+
+    # Load from PREMIS2
+    data = {"foo": "bar"}
+    atom._load_premis(
+        data, mets_document.get_file(file_uuid="ae8d4290-fe52-4954-b72a-0f591bee2e2f")
+    )
+    assert data == {
+        "foo": "bar",
+        "format_name": "JPEG 1.02",
+        "format_version": "1.02",
+        "format_registry_key": "fmt/44",
+        "format_registry_name": "PRONOM",
+        "size": "158131",
+    }
+
+    # Load from PREMIS3
+    data = {"foo": "bar"}
+    atom._load_premis(
+        data, mets_document.get_file(file_uuid="4583c44a-046c-4324-9584-345e8b8d82dd")
+    )
+    assert data == {
+        "foo": "bar",
+        "format_name": "Format name",
+        "format_version": "Format version",
+        "format_registry_key": "fmt/12345",
+        "format_registry_name": "PRONOM",
+        "size": "999999",
+    }
+
+
+@pytest.fixture
+def username():
+    return "test"
+
+
+@pytest.fixture
+def password():
+    return "test"
+
+
+def test_get_mets_unknown_mets(client, mocker, django_user_model, username, password):
+    django_user_model.objects.create_user(username=username, password=password)
+    client.login(username=username, password=password)
+    helpers.set_setting("dashboard_uuid", "test-uuid")
+    mocker.patch("components.archival_storage.views.aip_mets_file_download")
```

Comment:
You're mocking the actual view function here that I think is the target of these functional tests. This also makes the test module to fail when run in isolation in Docker. If I remove this line (and the pointer equivalent) from each test case the module passes in isolation.